dvc/repo/imp_url.py

```diff
@@ -27,6 +27,9 @@ def imp_url(
     ):
         url = relpath(url, wdir)
 
+    if not os.path.exists(wdir):
+        makedirs(wdir)
+
```

Comment:
Would it make sense to error-out if it doesn't exist? Kinda like `cp` and other tools do.

Reply:
Creating the directory if it doesn't exist feels like the more natural `import -o` behavior for me, but given that the current docs for `-o` say:
> If an existing directory is specified, the target data will be placed inside.
it would also be okay for us to error out here.

Reply:
please, don't forget to update the docs if we change this :)

Reply:
@pmrowla We also don't makedirs in `dvc run`, but rather raise an error that wdir doesn't exist, so it seems it would be more consistent if we raise and error here too. Same with `dvc add` that shares the same logic with `dvc run`. I wonder why that https://github.com/iterative/dvc/blob/35dd1fda9cfb4b645ae431f4621f2d1853d4fb7c/dvc/stage/utils.py#L27 check wasn't triggered here as well :thinking:

Reply:
The issue before was that we set `wdir` to `os.cwd()` if the output subdirectory didn't already exist, because we assumed that subdir not existing actually meant it was a remote URL and not a local path

Reply:
Behavior has been updated here so that we use the correct subdir as `wdir`, and then error out in `utils.check_stage_path` if it doesn't already exist